api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -86,6 +86,7 @@ message ExtAuthz {
 
   // Specifies a list of metadata namespaces whose values, if present, will be passed to the
   // ext_authz service. :ref:`filter_metadata <envoy_v3_api_field_config.core.v3.Metadata.filter_metadata>` is passed as an opaque ``protobuf::Struct``.
+  // If the same metadata key is found in both the request and connection metadata then the request value takes precedence.
```

Comment:
Please add links to the relevant metadata fields that could be forwarded (I'm assuming one is for route, and the other is for listener, but not sure).
This should be clarified in the line above ("list of metadata").

Reply:
I think this comment can be removed.

Reply:
hmm, I am not sure why this can be removed.  There is no code change between recent approvals.  In the implementation, the request value takes precedence over connection value. 

As I mentioned before. It is just namespace and i am not sure we can force extension to use different keys. Low level request metadata override high level connection metadata is similar to the pattern of per route config that router level overrides virtual host level.

We probably can add comment to suggest the best practice of avoiding key conflicts.

Reply:
You are correct that it is just a namespace and essentially can have any value.
However, the namespace should be well defined (and unique) by the filter that created it (probably [this doc](https://www.envoyproxy.io/docs/envoy/latest/configuration/advanced/well_known_dynamic_metadata) should be clarified, and some tests to ensure well-defined namespaces are used can be added).

Reply:
Thanks for the links. Yea I agree it will be great to be clarified in the doc around namespace. such as unique and well defined namespace is strongly encouraged(opposite probably should be disallowed)